ovirt/resource_ovirt_vm.go

```diff
@@ -438,7 +449,12 @@ func (c *providerContext) resourceOvirtVMCreate(d *schema.ResourceData, meta int
 	memoryPolicyBuilder := ovirtsdk4.NewMemoryPolicyBuilder()
 	if maximumMemory, ok := d.GetOk("maximum_memory"); ok {
 		// memory is specified in MB
-		memoryPolicyBuilder.Max(int64(maximumMemory.(int)) * int64(math.Pow(2, 20)))
+		memoryPolicyBuilder.Max(int64(maximumMemory.(int)) * BytesInMB)
+	}
+
+	if guaranteedMemory, ok := d.GetOk("guaranteed_memory"); ok {
+		//set guaranteed memory
+		memoryPolicyBuilder.Guaranteed(int64(guaranteedMemory.(int)) * BytesInMB)
```

Comment:
* Please also edit https://github.com/oVirt/terraform-provider-ovirt/pull/278/files#diff-a9ff694aa45bdc153e172e971927da1072cccf8ce4ef187ee2fbcddf9b81aebaR446 with BytesInMB.
* We should probably make sure guaranteedMemory and maximumMemory are only set when memory is set.
* Unrelated but, what is maximumMemory? I mean you give a machine some memory field, why would you need the MAX memory?

Reply:
maximum memory is for cases that we want to increase the Memory size while VM is running, this might affect the scheduling of the VM (the physical host) [0]


[0] https://www.ovirt.org/develop/release-management/features/virt/hot-plug-memory.html